java-connectors/kafka-connect-azure-eventhubs/src/main/java/io/lenses/java/streamreactor/connect/azure/eventhubs/source/BlockingQueueProducerProvider.java

```diff
@@ -56,7 +61,21 @@ public BlockingQueuedKafkaProducer createProducer(AzureEventHubsConfig azureEven
 
     KafkaConsumer<String, String> kafkaConsumer = new KafkaConsumer<>(consumerProperties);
 
-    return new BlockingQueuedKafkaProducer(topicPartitionOffsetProvider, recordBlockingQueue, kafkaConsumer, clientId, topic);
+    boolean shouldSeekToLatest = shouldConsumerSeekToLatest(azureEventHubsConfig);
+    String topic = azureEventHubsConfig.getString(AzureEventHubsConfigConstants.EVENTHUB_NAME);
+
+    return new BlockingQueuedKafkaProducer(topicPartitionOffsetProvider, recordBlockingQueue,
+        kafkaConsumer, clientId, topic, shouldSeekToLatest);
   }
 
+  private boolean shouldConsumerSeekToLatest(AzureEventHubsConfig azureEventHubsConfig) {
+    String seekValue = azureEventHubsConfig.getString(AzureEventHubsConfigConstants.CONSUMER_OFFSET);
+    if (EARLIEST_OFFSET.equals(seekValue)) {
```

Comment:
you should consider case insensitive check.

Reply:
Makes sense, I modified it, thanks!